app/views/shops/new.html.erb

```diff
@@ -0,0 +1,37 @@
+<h1>Input information for a new shop</h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for [@company, @shop], url:company_shops_path(@company), method: :post  do |shop_form | %>
```

Comment:
url:< space >company......
check spaces in this line, there are more places where it's wrong